source/hpack.c

```diff
@@ -668,7 +685,20 @@ int aws_hpack_insert_header(struct aws_hpack_context *context, const struct aws_
 
     /* Put the header at the "front" of the table */
     struct aws_http_header *table_header = s_dynamic_table_get(context, 0);
-    *table_header = *header;
+
+    /* TODO:: We can optimize this with ring buffer. */
+    /* allocate memory for the name and value, which will be deallocated whenever the entry is evicted from the table or
```

Comment:
Why not use aws_byte_buf for this?

Reply:
Yeah, I thought about that. But if we use aws_byte_buf, we will keep a list of aws_byte_buf to track which one should be cleaned up, as we just keep all those aws_byte_cursors.